src/Interpreters/TreeRewriter.cpp

```diff
@@ -1243,16 +1244,22 @@ void TreeRewriterResult::collectUsedColumns(const ASTPtr & query, bool is_select
         if (storage)
         {
             std::vector<String> hint_name{};
+            std::set<String> helper_hint_name{};
```

Comment:
Let's just use only one container, there is no point in having two of them
```suggestion
            std::set<String> hint_name{};
```

Reply:
I did an extra container just to check for duplicates and I didn't want to reorder the hints (as per https://github.com/ClickHouse/ClickHouse/issues/44130#issuecomment-1360066993).
Alternatively I could also do the following:
```cpp
auto hints = storage->getHints(name);
for (auto& hint : hints)
{
    if (std::find(std::begin(hint_name), std::end(hint_name), hint) == std::end(hint_name))
        hint_name.push_back(hint);
}
```

Reply:
Ok, but let's add a comment that we want to preserve the order